lib/ansible/module_utils/network/exos/facts/guest_virtual_machines/guest_virtual_machines.py

```diff
@@ -0,0 +1,116 @@
+based on the configuration.
+"""
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+import re
+from copy import deepcopy
+
+from ansible.module_utils.network.common import utils
+from ansible.module_utils.network.exos.argspec.guest_virtual_machines.guest_virtual_machines import Guest_virtual_machinesArgs
+from ansible.module_utils.network.exos.exos import send_requests
+
+
+class Guest_virtual_machinesFacts(object):
+    """ The exos guest_virtual_machines fact class
+    """
+
+    def __init__(self, module, subspec='config', options='options'):
+        self._module = module
+        self.argument_spec = Guest_virtual_machinesArgs.argument_spec
+        spec = deepcopy(self.argument_spec)
+        if subspec:
+            if options:
+                facts_argument_spec = spec[subspec][options]
+            else:
+                facts_argument_spec = spec[subspec]
+        else:
+            facts_argument_spec = spec
+
+        self.generated_spec = utils.generate_dict(facts_argument_spec)
+
+    def populate_facts(self, connection, ansible_facts, data=None):
+        """ Populate the facts for guest_virtual_machines
+        :param connection: the device connection
+        :param ansible_facts: Facts dictionary
+        :param data: previously collected conf
+        :rtype: dictionary
+        :returns: facts
+        """
+
+        if not data:
+            requests = [
+                {
+                    "path": "/rest/restconf/data/extreme-virtual-service:virtual-services-config/",
+                    "method": "GET"
+                },
+                {
+                    "path": "/rest/restconf/data/extreme-virtual-service:virtual-services-state/",
+                    "method": "GET"
+                }
+            ]
+
+            data = send_requests(self._module, requests=requests)
+
+        objs = []
+        if data:
+            conf_config = data[0]["extreme-virtual-service:virtual-services-config"]
+            conf_state = data[1]["extreme-virtual-service:virtual-services-state"]
+            if conf_config.get("virtual-service-config") and conf_state.get("virtual-service-state"):
+                conf_config = conf_config["virtual-service-config"]
+                conf_state = conf_state["virtual-service-state"]
+                for d in range(len(conf_config)):
+                    obj = self.render_config(self.generated_spec, conf_config[d], conf_state[d])
+                    if obj:
+                        objs.append(obj)
+
+        ansible_facts['ansible_network_resources'].pop('guest_virtual_machines', None)
+        facts = {}
+        if objs:
+            params = utils.validate_config(self.argument_spec, {'config': objs})
+            facts['guest_virtual_machines'] = params['config']
+
+        ansible_facts['ansible_network_resources'].update(facts)
+        return ansible_facts
+
+    def render_config(self, spec, conf1, conf2):
```

Comment:
It would be better avoiding changing method signature, you can pass the config and state as a dictionary to the method and unpack the them inside the render_config.